servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionFactoryOffloadingTest.java

```diff
@@ -0,0 +1,108 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class ConnectionFactoryOffloadingTest {
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testFactoryOffloading(boolean offload) throws Exception {
+        AtomicReference<Thread> factoryThread = new AtomicReference<>();
+
+        Thread appThread = Thread.currentThread();
+
+        try (ServerContext server = HttpServers.forPort(0)
+                .listenAndAwait(this::helloWorld)) {
+            SocketAddress serverAddress = server.listenAddress();
+
+            ConnectionFactoryFilter<SocketAddress, FilterableStreamingHttpConnection> factory =
+                    ConnectionFactoryFilter.withStrategy(original ->
+                                    new ConnectionFactory<SocketAddress, FilterableStreamingHttpConnection>() {
+                                private final ListenableAsyncCloseable close = emptyAsyncCloseable();
+
+                                @Override
+                                public Single<FilterableStreamingHttpConnection> newConnection(
+                                        final SocketAddress socketAddress, @Nullable final TransportObserver observer) {
+                                    factoryThread.set(Thread.currentThread());
+                                    return original.newConnection(socketAddress, observer);
+                                }
+
+                                @Override
+                                public Completable onClose() {
+                                    return close.onClose();
+                                }
+
+                                @Override
+                                public Completable closeAsync() {
+                                    return close.closeAsync();
+                                }
+
+                                @Override
+                                public Completable closeAsyncGracefully() {
+                                    return close.closeAsyncGracefully();
+                                }
+                            },
+                            offload ? ConnectExecutionStrategy.offload() : ConnectExecutionStrategy.anyStrategy());
+
+            try (HttpClient client = HttpClients.forResolvedAddress(serverAddress)
+                    .appendConnectionFactoryFilter(factory)
+                    .build()) {
+                Single<HttpResponse> single = client.request(client.get("/sayHello"));
+                HttpResponse response = single.toFuture().get();
+                assertThat("unexpected status", response.status(), is(HttpResponseStatus.OK));
+            }
+        }
+        assertTrue((offload && !IoThreadFactory.IoThread.isIoThread(factoryThread.get())) ||
+                        (!offload && factoryThread.get() == appThread),
+                "incorrect offloading, offload=" + offload + " thread=" + factoryThread.get());
+    }
+
+    private Single<HttpResponse> helloWorld(HttpServiceContext ctx,
```

Comment:
do we have a test that tests `ConnectionFactoryFilter` ability to return `HttpExecutionStrategy` and `ConnectExecutionStrategy + HttpExecutionStrategy`? if not can you plz add one as the merging/combining is relatively complex internally so good to verify we don't break it in the future.

Reply:
Now added. The `ConnectionFactoryOffloadingTest` now exercises the use of `ConnectAndHttpExecutionStrategy`.